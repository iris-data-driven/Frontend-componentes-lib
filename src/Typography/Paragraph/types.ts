type ParagraphTypes = 'p1' | 'p2' | 'p3' | 'p4';

export type ParagraphProps = {
  type?: ParagraphTypes;
  className?: string;
  onClick?: () => void;
};
