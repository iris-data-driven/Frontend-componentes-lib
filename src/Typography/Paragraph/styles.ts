import styled from 'styled-components';

import texts from 'styles/texts';

import { ParagraphProps } from './types';

export const Wrapper = styled.p<ParagraphProps>`
  ${props => texts.paragraphs[props.type]}
`;
