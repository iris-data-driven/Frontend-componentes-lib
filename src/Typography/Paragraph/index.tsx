import React, { FC } from 'react';

import { Wrapper } from './styles';

import { ParagraphProps } from './types';

const Paragraph: FC<ParagraphProps> = ({ type, children }) => (
  <Wrapper data-testid="paragraph" type={type}>
    {children}
  </Wrapper>
);

export default Paragraph;
