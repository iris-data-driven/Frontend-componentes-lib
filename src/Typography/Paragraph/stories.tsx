import React from 'react';

import Paragraph from './';

export default {
  title: 'Typography|Paragraph',
};

export const Paragraph1 = () => <Paragraph type="p1">Paragraph 1</Paragraph>;
export const Paragraph2 = () => <Paragraph type="p2">Paragraph 2</Paragraph>;
export const Paragraph3 = () => <Paragraph type="p3">Paragraph 3</Paragraph>;
export const Paragraph4 = () => <Paragraph type="p4">Paragraph 4</Paragraph>;
