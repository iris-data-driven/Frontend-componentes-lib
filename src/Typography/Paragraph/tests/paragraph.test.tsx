import React from 'react';
import { render, screen } from '@testing-library/react';

import Paragraph from './';

describe('Paragraph', () => {
  it('Should render the paragraph with the correct text and style', () => {
    const text = 'This is a paragraph type 1';

    render(<Paragraph type="p1">{text}</Paragraph>);

    const component = screen.getByRole('paragraph');
    expect(component).toMatchSnapshot();
    expect(component).toHaveTextContent(text);
  });

  it('Should render the paragraph with the correct text and style', () => {
    const text = 'This is a paragraph type 2';

    render(<Paragraph type="p2">{text}</Paragraph>);

    const component = screen.getByRole('paragraph');
    expect(component).toMatchSnapshot();
    expect(component).toHaveTextContent(text);
  });

  it('Should render the paragraph with the correct text and style', () => {
    const text = 'This is a paragraph type 3';

    render(<Paragraph type="p3">{text}</Paragraph>);

    const component = screen.getByRole('paragraph');
    expect(component).toMatchSnapshot();
    expect(component).toHaveTextContent(text);
  });

  it('Should render the paragraph with the correct text and style', () => {
    const text = 'This is a paragraph type 4';

    render(<Paragraph type="p4">{text}</Paragraph>);

    const component = screen.getByRole('paragraph');
    expect(component).toMatchSnapshot();
    expect(component).toHaveTextContent(text);
  });
});
