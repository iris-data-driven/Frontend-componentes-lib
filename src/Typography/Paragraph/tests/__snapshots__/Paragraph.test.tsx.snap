// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Should render the paragraph with the correct text and style 1`] = `
<p
  class="sc-AxjAm iAGhph"
  role="paragraph"
  type="p1"
>
  This is a paragraph type 1
</p>
`;

exports[`Paragraph Should render the paragraph with the correct text and style 2`] = `
<p
  class="sc-AxjAm cllMKu"
  role="paragraph"
  type="p2"
>
  This is a paragraph type 2
</p>
`;

exports[`Paragraph Should render the paragraph with the correct text and style 3`] = `
<p
  class="sc-AxjAm fUUqhx"
  role="paragraph"
  type="p3"
>
  This is a paragraph type 3
</p>
`;

exports[`Paragraph Should render the paragraph with the correct text and style 4`] = `
<p
  class="sc-AxjAm eqfLW"
  role="paragraph"
  type="p4"
>
  This is a paragraph type 4
</p>
`;
